feat: deprecate `Frame.prototype.name`

jrandolf-google opened the pull request:
`Frame.prototype.name` has a lot of problem associated with it. See https://github.com/puppeteer/puppeteer/issues/12083.

We resolve the use-case with the new `frameElement` getter which returns the frame element associated with a given frame.

Fixes https://github.com/puppeteer/puppeteer/issues/12083

Lightning00Blade reviewed (approved):
LGTM, but I think we should remove the uses from test as well, WDYT?

sadym-chromium commented on packages/puppeteer-core/src/api/Frame.ts:
```diff
@@ -405,7 +405,7 @@ export abstract class Frame extends EventEmitter<FrameEvents> {
 
   /**
-   * @internal
+   * @returns The frame element associated with this frame (if any).
```
Please add tests for `frameElement()`, as it became a public API

jrandolf committed:
feat: deprecate `Frame.prototype.name`

`Frame.prototype.name` has a lot of problem associated with it. See https://github.com/puppeteer/puppeteer/issues/12083.

We resolve the use-case with the new `frameElement` getter which returns the frame element associated with a given frame.

Fixes https://github.com/puppeteer/puppeteer/issues/12083

Lightning00Blade commented on test/src/launcher.spec.ts:
```diff
@@ -777,7 +777,7 @@ describe('Launcher specs', function () {
         }
       });
-      it('should be able to reconnect to a disconnected browser', async () => {
+      it.skip('should be able to reconnect to a disconnected browser', async () => {
```
Why this change? Event if we skip test we do it via Expectations.

Lightning00Blade commented on docs/api/puppeteer.frame.name.md:
````diff
@@ -4,6 +4,15 @@ sidebar_label: Frame.name
+>
+> ```ts
+> const element = await frame.frameElement();
+> const name = await element.evaluate(frame => frame.name);
````
Let's provide an example with the `id` as `frame.name()` would also return the Id of the element.
